Sources/BraveWallet/Crypto/Portfolio/PortfolioView.swift

```diff
@@ -213,6 +176,174 @@ struct PortfolioView: View {
       self.collectionViewRef = .init(collectionView)
     }
   }
+  
+  /// Header for the assets section(s) containing the title, edit user assets and filter buttons
+  private var assetSectionsHeader: some View {
+    HStack {
+      Text(Strings.Wallet.assetsTitle)
+      if portfolioStore.isLoadingDiscoverAssets {
+        ProgressView()
+          .padding(.leading, 5)
+      }
+      Spacer()
+      editUserAssetsButton
+        .padding(.trailing, 10)
+      filtersButton
+    }
+    .textCase(nil)
+    .padding(.horizontal, -8)
+    .frame(maxWidth: .infinity, alignment: .leading)
+  }
+  
+  private var emptyAssetsState: some View {
+    VStack(spacing: 10) {
+      Image("portfolio-empty", bundle: .module)
+        .aspectRatio(contentMode: .fit)
+      Text(Strings.Wallet.portfolioEmptyStateTitle)
+        .font(.headline)
+        .foregroundColor(Color(WalletV2Design.textPrimary))
+      Text(Strings.Wallet.portfolioEmptyStateDescription)
+        .font(.footnote)
+        .foregroundColor(Color(WalletV2Design.textSecondary))
+    }
+    .multilineTextAlignment(.center)
+    .padding(.vertical)
+  }
+  
+  /// Grouped / Ungrouped asset `Section`s for the List
+  @ViewBuilder private var assetSections: some View {
+    if portfolioStore.isShowingAssetsLoadingState || portfolioStore.isShowingAssetsEmptyState {
+      Section(content: {
+        Group {
+          if portfolioStore.isShowingAssetsLoadingState {
+            SkeletonLoadingAssetView()
+          } else { // isShowingAssetsEmptyState
+            emptyAssetsState
+          }
+        }
+        .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      }, header: {
+        assetSectionsHeader
+      })
+    } else {
```

Comment:
can't we do if/else if?

Reply:
I'm not sure what you mean here? Adding `else if` here instead would just be unnecessary code:
```
if portfolioStore.isShowingAssetsLoadingState || portfolioStore.isShowingAssetsEmptyState {
...
} else if !portfolioStore.isShowingAssetsLoadingState && !portfolioStore.isShowingAssetsEmptyState {
..
}
```